checkbox-ng/checkbox_ng/resume_menu.py

```diff
@@ -0,0 +1,350 @@
+            chosen_session = self._entries[self.focused_index][0]
+            if chosen_session != self._chosen_session:
+                self.chosen_session = chosen_session
+            # now let's show action menu, operator will chose what to do with
+            # the session
+            self.loop.widget = self._action_view
+
+        elif key == "esc":
+            self.chosen_session = None
+
+            raise urwid.ExitMainLoop()
+        elif key in ["d", "D"]:
+            # user chose to delete the session
+            self.chosen_session = self._entries[self.focused_index][0]
+            self._chosen_action = "delete"
+            raise urwid.ExitMainLoop()
+
+    def _handle_input_on_action_menu(self, key):
+        """Handle input on the action menu."""
+        if key == "esc":
+            # user cancelled the action, go back to the session selection
+            self.chosen_session = None
+            self.loop.widget = self._body
+
+        elif key == "enter":
+            # user chose an action, let's record it and exit
+            # if the action is "comment" we need to show the comment box
+
+            action_index = self._action_menu.base_widget.focus_position - len(
+                self._ACTION_MENU_STATIC_ELEMENTS
+            )
+            action = self._action_buttons[action_index][1]
+
+            self._chosen_action = action
+            if action == "comment":
+                self.loop.widget = self._comment_view
+            else:
+                raise urwid.ExitMainLoop()
+
+    def _handle_input_on_comment_box(self, key):
+        """Handle input on the comment box."""
+        if key == "esc":
+            self._comment_edit_box.edit_text = self._comment
+            self._chosen_action = None
+            self.loop.widget = self._action_view
+        if key == "enter":
+            self._comment = self._comment_edit_box.edit_text
+            self._chosen_action = None
+            self.loop.widget = self._action_view
+
+    def _unhandled_input(self, key):
+        """
+        Handle all input in the urwid loop for the resume screen.
+        This propagates the input to the correct handler depending on
+        the current state of the menu.
+        """
+        if self._chosen_session:
+            # if the session is already chosen it means we're
+            # on the action selection menu or entering a comment
+            if self._chosen_action:
+                self._handle_input_on_comment_box(key)
+            else:
+                self._handle_input_on_action_menu(key)
+        else:
+            # otherwise it means were' at the session selection menu
+            self._handle_input_on_session_menu(key)
+
+    @property
+    def focused_index(self):
+        """
+        Return the index of the currently focused session.
+        """
+        # we need to substract the number of static elements, for instance,
+        # if there's 5 static elements, the first session item will have an
```

Comment:
I don't get what a static element is/where does it come from from this comment alone. I think a better name for them is `STATIC_MENU_UI_ELEMENTS` or something like that. Also: does adding them to the menu mean that they are selectable? If I select them does it the `max(0,` make me select the first "real" menu entity?

Reply:
Added more explanation in the comment.
As for select-ability of those static items - no, they can't be selected. But the index is initiated with 0, this is why I'm doing the max(0...) thingy.